From: Felix Krull <f_krull@gmx.de>
Date: Fri, 27 Apr 2018 08:06:21 -0700
Subject: issue4961

# DP: Issue #4961: Inconsistent/wrong result of askyesno function in
# DP: tkMessageBox with Tcl8.5.
---
 Lib/lib-tk/tkMessageBox.py | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/Lib/lib-tk/tkMessageBox.py b/Lib/lib-tk/tkMessageBox.py
index 2c1c1e0..e8ee23e 100644
--- a/Lib/lib-tk/tkMessageBox.py
+++ b/Lib/lib-tk/tkMessageBox.py
@@ -73,11 +73,13 @@ def _show(title=None, message=None, icon=None, type=None, **options):
     if title:   options["title"] = title
     if message: options["message"] = message
     res = Message(**options).show()
-    # In some Tcl installations, Tcl converts yes/no into a boolean
+    # In some Tcl installations, yes/no is converted into a boolean.
     if isinstance(res, bool):
-        if res: return YES
+        if res:
+            return YES
         return NO
-    return res
+    # In others we get a Tcl_Obj.
+    return str(res)
 
 def showinfo(title=None, message=None, **options):
     "Show an info message"
